code/modules/reagents/reagent_containers/food/snacks.dm

```diff
@@ -1112,6 +1124,19 @@
 	..()
 	reagents.clear_reagents()
 
+/obj/item/weapon/reagent_containers/food/snacks/pie/caramelpie
+	name = "caramel pie"
+	desc = "Sweet pie made with caramel"
+	icon_state = "pie"
+	trash = /obj/item/trash/pietin
+	food_flags = FOOD_SWEET
+
+/obj/item/weapon/reagent_containers/food/snacks/pie/caramelpie/New()
+	..()
+	reagents.add_reagent(NUTRIMENT, 4)
```

Comment:
put `reagents.clear_reagents()` right before this

Reply:
Added, testing now